app/src/main/java/com/mk/tasky/agenda/data/remote/worker/SyncLocalWithRemoteWorker.kt

```diff
@@ -0,0 +1,112 @@
+package com.mk.tasky.agenda.data.remote.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.mk.tasky.agenda.data.local.AgendaDao
+import com.mk.tasky.agenda.data.local.entity.relations.EventAttendeesCrossRef
+import com.mk.tasky.agenda.data.mapper.toDomain
+import com.mk.tasky.agenda.data.mapper.toDto
+import com.mk.tasky.agenda.data.mapper.toEntity
+import com.mk.tasky.agenda.data.remote.AgendaApi
+import com.mk.tasky.agenda.domain.alarm.AlarmRegister
+import com.mk.tasky.agenda.domain.model.AgendaItem
+import com.mk.tasky.agenda.domain.model.SyncItem
+import com.mk.tasky.agenda.domain.model.SyncType
+import com.mk.tasky.agenda.domain.repository.AgendaRepository
+import com.mk.tasky.agenda.domain.uploader.EventUploader
+import com.mk.tasky.agenda.util.toLong
+import com.mk.tasky.core.data.util.resultOf
+import com.mk.tasky.core.util.AgendaItemType
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+import java.lang.Exception
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.ZoneId
+import java.util.*
+
+
+@HiltWorker
+class SyncLocalWithRemoteWorker @AssistedInject constructor(
+    @Assisted val context: Context,
+    @Assisted val workerParameters: WorkerParameters,
+    private val api: AgendaApi,
+    private val dao: AgendaDao,
+    private val alarmRegister: AlarmRegister
+) : CoroutineWorker(context, workerParameters) {
+    override suspend fun doWork(): Result {
+        val agenda = getAgenda()
+        val agendaDto = agenda.getOrNull()
+        if (agenda.isFailure || agendaDto == null) return Result.retry()
```

Comment:
It could make sense to check what the error is because there are errors where you don't want to retry, such as Unauthorized

Reply:
Will handle all type of errors in the future. But you are right